obs/project.go

```diff
@@ -91,3 +97,102 @@ type RepositoryPath struct {
 	Project    string `json:"project" xml:"project,attr"`
 	Repository string `json:"repository" xml:"repository,attr"`
 }
+
+func (c Client) CreateProject(project *Project) error {
+	xmlData, err := xml.MarshalIndent(project, "", " ")
+	if err != nil {
+		return err
+	}
+
+	urlPath, err := url.JoinPath(c.APIURL, "source", project.Name, "_meta")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req, err := http.NewRequest("PUT", urlPath, bytes.NewBuffer(xmlData))
+	if err != nil {
+		return &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("failed to generate a new request: %v", err),
```

Comment:
We tend to use [gerund](https://www.thesaurus.com/e/grammar/whats-a-gerund/) for our error messages. For example:

```suggestion
			Message:        fmt.Sprintf("creating obs project: creating request: %v", err),
```